src/library/scala/collection/mutable/OpenHashMap.scala

```diff
@@ -64,7 +72,22 @@ extends AbstractMap[Key, Value]
   private[this] val actualInitialSize = OpenHashMap.nextPositivePowerOfTwo(initialSize)
 
   private var mask = actualInitialSize - 1
-  private var table : Array[Entry] = new Array[Entry](actualInitialSize)
+
+  /** The state of each of the slots in the hash table.
+    * One of `Empty`, `Deleted` or `Occupied`.
+    */
+  private[this] var slotStates = new Array[Byte](actualInitialSize)
```

Comment:
I updated the performance graphs in the JIRA ticket. It's gotten a bit better, and a bit worse.

What bothers me is why the `Some` isn't being elided with the `getOrElse(0)` call. It's sticks around also if I use the `AbstractMap.getOrElse(i, 0)`. Building tests with `-Yopt:l:classpath`.